packages/moonstone/styles/internal/fonts.less

```diff
@@ -33,20 +34,20 @@
 // In the case of glyphs existing in two fonts, the later glyphs will be used.
 @fonts: {
 	am: local("LG Smart UI Amharic");
-	as: local("LG Smart UI Bengali"); // same as `bn`
-	bn: local("LG Smart UI Bengali");
-	en-JP: local("LG Smart UI JP"), local("LG Display_JP_Bold") 700; // same as `ja`
-	gu: local("LG Smart UI Gujarati");
+	ar: local("LG Smart UI AR HE TH") normal 300 600 700;
+	as: local("LG Smart UI Bengali") normal 300 600 700; // same as `bn`
+	bn: local("LG Smart UI Bengali") normal 300 600 700;
+	gu: local("LG Smart UI Gujarati") normal 300 600 700;
+	he: local("LG Smart UI AR HE TH") normal 300 600 700;
 	hi: local("LG Smart UI Devanagari");
-	ja: local("LG Smart UI JP"), local("LG Display_JP_Bold") 700;
 	kn: local("LG Smart UI Kannada");
-	ml: local("LG Smart UI Malayalam");
+	ml: local("LG Smart UI Malayalam") normal 300 600 700;
 	mr: local("LG Smart UI Devanagari"); // same as `hi`
 	or: local("LG Smart UI Oriya");
-	pa: local("LG Smart UI Gurmukhi");
-	ta: local("LG Smart UI Tamil");
+	pa: local("LG Smart UI Gurmukhi") normal 300 600 700;
+	ta: local("LG Smart UI Tamil") normal 300 600 700;
 	te: local("LG Smart UI Telugu");
-	ur: local("LG Smart UI Urdu") normal 300 600 700;
```

Comment:
Please do not remove `ur`.

Reply:
If `ar` is handled in this language block for applying `LG Smart UI AR HE TH`, `ur` locales need to be handled in `fontGenerator.js` and in that case, this line is useless.

Reply:
`ar` should not be handled in this language definition block.

Reply:
`non-latin` is ambiguous and it has just three locales (`ar`, `he`, and `th`). Why do you think they should not be handled in this block?

Reply:
Correct, "non-latin" is no longer representative of what is inside that font. But these are just the variable names; the names have no impact on *how* they are used in this file.

I've explained the reasoning in the main level of this PR's comments. https://github.com/enactjs/enact/pull/2556#pullrequestreview-289452667

Reply:
So what about the case of `as`, `bn` and `zh-HK`, `zh-TW`. These are similarly repeated. Because "non-latin" is no longer representative of what is inside that font, I think precedence is not important things.

Reply:
Precedence in CSS is always important. It's how the language works and how everything is structured. Based on the very definition of the word precedence, these rules literally precede the later rules. They establish a base for this font to be applied correctly.

The `as`, `bn`, `zh-HK`, and `zh-TW` lines are not repeated because this block of code does multiple jobs:
* It creates locale/language-specific font-families for apps to refer to directly
* Adds the glyphs of the font to the stack of glyphs that comprise the "Moonstone" font-family.

`ar`, `he`, and `th` are **not** included here because the font that contains those glyphs includes **multiple languages** and isn't specific to that language. Therefore it gets treated differently. The way Moonstone treats these cumulative fonts is by referring to them in the "non-Latin" category and `@font-system-src-non-latin` variable. Please do not get confused by the fact that the variable name includes the words "non-latin". This is a legacy variable name and can't change now. We just have to deal with the fact that it has those words in the its name, however, this is completely irrelevant to *how* the variable is being used. It is establishing a basis for the multi-lingual "Moonstone" font-family in the rules you've removed below. Those rules must be restored.

Reply:
ar, he, and th do same thing that you commented about as, bn and zhs doing(creates font-faces and add the glyphs).
If multiple languages(arabic, thai, hebrew) is printed same time, it's OK because they use same font.
Strictly speaking, LG Smart UI JP has multiple language, too.
Yes, precedence in CSS and lagacy code are always important. But you've already changed the legacy codes through your overhaul PR. In this state, you never get LG Smart UI AR HE TH in ar locale. In all ar locales, LG Smart UI UR is printed now. If you would like to keep your font theme in Moonstone, it's OK, but I would like to you fix these bugs ASAP. If you do so, I will close this PR without merging.

Reply:
> ar, he, and th do same thing that you commented about as, bn and zhs doing(creates font-faces and add the glyphs).

Yes, however `LG Smart UI AR HE TH` is a cumulitive font that includes multiple languages. I'm content with retaining the ar, he, and th locales in this collection. Having them generate a standalone font-family is fine, even if it contains other languages, because at least it will cover the target language too.

>But you've already changed the legacy codes through your overhaul PR.

This was before we released 3.0.0, when breaking changes were acceptable. Now they're not, until 4.0.0. So we must live with this variable name until then.

>In this state, you never get LG Smart UI AR HE TH in ar locale. In all ar locales, LG Smart UI Urdu is printed now.

This is because the Urdu font has conflicting characters with the `LG Smart UI AR HE TH` font. The last one applied gets to stay. The Urdu font is likely displaying because the "legacy" rules don't specifically call-out "normal" for their font weight, so this rule takes precedence over the one below it, since it's more specific. Adding "normal" to the legacy ruleset will resolve this and will mean that we don't break an app that refers to `font-family: "Moonstone ur";`.

> ...I would like to you fix these bugs ASAP. If you do so, I will close this PR without merging.

I am interested in keeping and using this PR as it does include good changes and fixes. I was hoping, through discussion, that we could make changes/updates to parts of it and I could simply approve it, rather than redoing the work all on my own.

Reply:
>> This is because the Urdu font has conflicting characters with the LG Smart UI AR HE TH font. The last one applied gets to stay. The Urdu font is likely displaying because the "legacy" rules don't specifically call-out "normal" for their font weight, so this rule takes precedence over the one below it, since it's more specific. Adding "normal" to the legacy ruleset will resolve this and will mean that we don't break an app that refers to font-family: "Moonstone ur";.

Does you mean like this?
```
@font-system-src-non-latin-400:  local("LG Smart UI AR HE TH Regular") normal, local("LGSmartUIARHETH-Regular") normal;

Reply:
Technically this won't work since this variable just holds multiple font file references, not weights. We'll need to add the `normal` keyword down below, in the first line of this section:

```
/* ----- Manual Moonstone Locale Fonts (Stacking onto "Moonstone") ------ */
.buildFontFace(@font-family-base-name; @font-system-src-non-latin-400, normal);
.buildFontFace(@font-family-base-name; @font-system-src-non-latin-300; 300);
.buildFontFace(@font-family-base-name; @font-system-src-non-latin-600; 600);
.buildFontFace(@font-family-base-name; @font-system-src-non-latin-700; 700);
```